### PR TITLE
Add auth token to ClientOptions

### PR DESCRIPTION
**Pull request by ChiChou**
Support frida-server `--token` option

### CODE REVIEW

**Comment by ChiChou** [comment]
Looks like https://github.com/frida/node-dbus-next only used the parser logic from upstream.
The upstream repo looks dead, I would like to contribute a library for the same api but in typescript

**Comment by oleavr** [comment]
Thanks! That would be awesome by the way.

I accidentally un-drafted this -- please let me know when it's ready to be merged 😊

**Comment by ChiChou** [comment]
🙌 for this PR it's ready to be merged, thanks

I am planning to port more interfaces. That's gonna be more future work

File: lib/client.ts [hunk 1]
```diff
@@ -38,6 +39,10 @@ export class Client {
                 break;
         }
         this._serverUrl = `${scheme}://${host}/ws`;
+
+        if (typeof options.token === "string") {
```
Comment (oleavr) [inline lib/client.ts]:
```suggestion
        if (options.token !== undefined) {
```

Would be good to use a strict check here.

**Review by oleavr — CHANGES_REQUESTED** [review]
Thanks! :raised_hands: (And apologies for the delay here!)

**Comment by oleavr** [comment]
Just released this in `@frida/web-client@3.1.0` :tada:

**Pull request merged**